src/Components/TodoApp.js

```diff
@@ -0,0 +1,113 @@
+
+import TodoItem from './TodoItem';
+import Input from "./Input";
+import Footer from "./Footer";
+import Button from "./Button";
+import { useCallback, useMemo } from "react";
+import Box from "@mui/material/Box";
+import { useSelector, useDispatch } from "react-redux";
+import {
+  addTodo,
+  checkAll,
+  toggleDone,
+  edit,
+  replace,
+  deleteTodo,
+  clear,
+  newTodo,
+  editTodo,
+  clickFilter,
+} from "./actions/index";
+
+function TodoApp(){
+
+   
+    const initial = useSelector((state) => state.todoReducers);
+    const { all, mode, newToDo, editToDo } = initial;
+  
+    //to dispatch action
+    const dispatch = useDispatch();
+  
+    //giving filtered array
+    const { listToMap, list, completed, active } = useMemo(() => {
+      const activeArray = [...all].filter((item) => item.done === false);
+      const completeArray = [...all].filter((item) => item.done === true);
+  
+      return {
+        listToMap:
+          mode === "active"
+            ? activeArray
+            : mode === "completed"
+            ? completeArray
+            : all,
+        list: all.length,
+        completed: completeArray.length,
+        active: activeArray.length,
+      };
+    }, [mode, all]);
+  
+    //for adding todo
+    const addItem = useCallback(
+      (e, newToDo) => {
+        e.key === "Enter" && dispatch(addTodo(newToDo));
+      },
+      [dispatch]
+    );
+  
+    //for edit input
+    const editItem = useCallback(
+      (e, id, editToDo) => {
+        e.key === "Enter" && dispatch(edit(id, editToDo));
+      },
+      [dispatch]
+    );
+  
+    //to replace todo
+    const replaceItem = useCallback(
+      (id, action) => dispatch(replace(id, action)),
+      [dispatch]
+    );
+  
+    //to complete all todos
+    const clickAll = useCallback(
+      () => dispatch(checkAll(completed, list)),
+      [dispatch, completed, list]
+    );
+  
+    return (
+      <Box>
+        <h1>todos</h1>
+        <div className="main">
+        <div className="top">
+          <Button countAll={list} checkAll={clickAll} />
+          <Input 
+            value={newToDo}
+            change={(e) => dispatch(newTodo(e.target.value))}
+            enter={(e) => addItem(e, newToDo)}
```

Comment:
why do you need to put newTodo in the arguments if it already exists in the reducer?